tests/framework/ngf.go

```diff
@@ -0,0 +1,188 @@
+
+	if output, err := exec.Command("kubectl", "apply", "-f", apiPath).CombinedOutput(); err != nil {
+		return output, err
+	}
+
+	if webhookRequired(k8sVersion) {
+		webhookPath := fmt.Sprintf("%s/v%s/webhook-install.yaml", gwInstallBasePath, apiVersion)
+
+		if output, err := exec.Command("kubectl", "apply", "-f", webhookPath).CombinedOutput(); err != nil {
+			return output, err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		if err := wait.PollUntilContextCancel(
+			ctx,
+			500*time.Millisecond,
+			true, /* poll immediately */
+			func(ctx context.Context) (bool, error) {
+				var deployment apps.Deployment
+				key := types.NamespacedName{
+					Namespace: "gateway-system",
+					Name:      "gateway-api-admission-server",
+				}
+
+				if err := k8sClient.Get(ctx, key, &deployment); err != nil {
+					return false, err
+				}
+
+				if deployment.Status.ReadyReplicas == 1 {
+					return true, nil
+				}
+
+				return false, nil
+			},
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	return nil, nil
+}
+
+// InstallNGF installs NGF.
+func InstallNGF(cfg InstallationConfig, extraArgs ...string) ([]byte, error) {
+	args := []string{
+		"install", cfg.ReleaseName, cfg.ChartPath, "--create-namespace", "--namespace", cfg.Namespace, "--wait",
+	}
+
+	if cfg.NgfImageRepository != "" {
+		args = append(args, formatValueSet("nginxGateway.image.repository", cfg.NgfImageRepository)...)
+		if cfg.ImageTag != "" {
+			args = append(args, formatValueSet("nginxGateway.image.tag", cfg.ImageTag)...)
+		}
+		if cfg.ImagePullPolicy != "" {
+			args = append(args, formatValueSet("nginxGateway.image.pullPolicy", cfg.ImagePullPolicy)...)
+		}
+	}
+
+	if cfg.NginxImageRepository != "" {
+		args = append(args, formatValueSet("nginx.image.repository", cfg.NginxImageRepository)...)
+		if cfg.ImageTag != "" {
+			args = append(args, formatValueSet("nginx.image.tag", cfg.ImageTag)...)
+		}
+		if cfg.ImagePullPolicy != "" {
+			args = append(args, formatValueSet("nginx.image.pullPolicy", cfg.ImagePullPolicy)...)
+		}
+	}
+
+	if cfg.ServiceType != "" {
+		args = append(args, formatValueSet("service.type", cfg.ServiceType)...)
+	}
+
+	fullArgs := append(args, extraArgs...)
+
+	return exec.Command("helm", fullArgs...).CombinedOutput()
+}
+
+// UninstallNGF uninstalls NGF.
+func UninstallNGF(cfg InstallationConfig, k8sClient client.Client) ([]byte, error) {
+	args := []string{
+		"uninstall", cfg.ReleaseName, "--namespace", cfg.Namespace,
+	}
+
+	output, err := exec.Command("helm", args...).CombinedOutput()
+	if err != nil {
+		return output, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	err = k8sClient.Delete(ctx, &core.Namespace{ObjectMeta: metav1.ObjectMeta{Name: cfg.Namespace}})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	var crList apiext.CustomResourceDefinitionList
```

Comment:
Why include this functionality in the uninstall NGF function? There might be some cases where we don't want to delete all the Gateway API resources we created for the tests. 

Maybe we could extract this into a separate function that can be called from the test.

Reply:
This is just for our CRDs, not Gateway API.

Reply:
Ah, I see. I misread the group name.